CLIENTS.md

```diff
@@ -0,0 +1,369 @@
+            }
+        }
+    }
+}
+
+// Request with a self-managed key
+{
+    "hashedRekordRequestV0_0_2": {
+        "digest": "<base64 digest of artifact>",
+        "signature": {
+            "content": "<base64 signature>",
+            "verifier": {
+                "publicKey": {
+                    "rawBytes": "<base64 DER-encoded public key>"
+                },
+                "keyDetails": "PKIX_ECDSA_P256_SHA_256"
+            }
+        }
+    }
+}
+
+// Request with an attestation
+{
+    "dsseRequestV0_0_2": {
+        "envelope": {
+            "payload": "<base64-encoded message>",
+            "payloadType": "<type, e.g. application/vnd.in-toto+json>",
+            "signatures": [
+                {
+                    "sig": "<base64-encoded signature>",
+                    "keyid": ""
+                }
+            ]
+        },
+        "verifier": {
+            "x509Certificate": "<base64 DER-encoded certificate>",
+            // Must match signing algorithm
+            "keyDetails": "PKIX_ECDSA_P256_SHA_256"
+        }
+    }
+}
+
+// Request with an attestation with a self-managed key
+{
+    "dsseRequestV0_0_2": {
+        "envelope": {
+            "payload": "<base64-encoded message>",
+            "payloadType": "<type, e.g. application/vnd.in-toto+json>",
+            "signatures": [
+                {
+                    "sig": "<base64-encoded signature>",
+                    "keyid": ""
+                }
+            ]
+        },
+        "verifier": {
+            "publicKey": {
+                "rawBytes": "<base64 DER-encoded public key>"
+            },
+            // Must match signing algorithm
+            "keyDetails": "PKIX_ECDSA_P256_SHA_256"
+        }
+    }
+}
+```
+
+The response will be a
+[`TransparencyLogEntry` message](https://github.com/sigstore/protobuf-specs/blob/5296f13d62e7fad428581d969f664c30cc52f549/protos/sigstore_rekor.proto#L94),
+which should be persisted in a bundle. Clients no longer need to transform the
+Rekor response into a `TLE` message to store in the bundle.
+
+### Two Entry Types
+
+Rekor v2 only supports `hashedrekord` (`HashedRekordRequestV0_0_2`) and
+`dsse` (`DSSERequestV0_0_2`) entry types, dropping a number of unused types
+such as `jar`, `alpine`, `rpm`, and the older types `rekord` and `intoto`.
+Additional types may be added in the future if there is demand, but this
+will require updating the client specification so that all clients implement
+support for these types.
+
+### Certificate and Public Key Verifiers
+
+Rekor v2 only supports signature verification using a certificate or a
+public key, dropping support for PGP, minisign, pkcs7, SSH and TUF.
+Additional verifiers may be added in the future, but this will also require
+updating the client specification.
+
+### Handling Longer Requests
+
+Clients need to increase request timeouts to at least 10 seconds.
+
+Rekor now batches uploads so that checkpoints are published less frequently.
```

Comment:
Not that this matters to clients but this is technically not true, the checkpoint publishing interval is not related to the batch interval.

Reply:
Thanks, removed the mention of batching and clarified that this is just about blocking until a checkpoint has been published.